go/engine/login_provisioned_device.go

```diff
@@ -173,6 +173,16 @@ func (e *LoginProvisionedDevice) run(ctx *Context) error {
 
 func (e *LoginProvisionedDevice) unlockDeviceKeys(ctx *Context, me *libkb.User) error {
 
+	// CORE-5876 idea that lksec will be unusable if reachablity state is NO
+	// and the user changed passphrase with a different device since it won't
+	// be able to sync the new server half.
+	if e.G().ConnectivityMonitor.IsConnected(ctx.NetContext) != libkb.ConnectivityMonitorYes {
+		e.G().Log.Debug("LoginProvisionedDevice: in unlockDeviceKeys, ConnectivityMonitor says not reachable, check to make sure")
+		if err := e.G().ConnectivityMonitor.CheckReachability(ctx.NetContext); err != nil {
+			e.G().Log.Debug("error checking reachability: %s", err)
+		}
+	}
+
```

Comment:
I would expect an error out here if not reachable, no?

Reply:
Well, it only ever returns nil, so this doesn't really matter, but it's fine to keep going without syncing the server half most of the time, since passphrase changes are rare.

Reply:
Maybe we could do a followup here on `IsConnected` to report if it changed? Juts for logging purposes.